lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -7870,6 +7870,17 @@ ActorReferenceResult ActorReferenceResult::forReference(
         (!actorInstance || actorInstance->isSelf())) {
       auto type = fromDC->mapTypeIntoContext(decl->getInterfaceType());
       if (!type->isSendableType()) {
+        // If we have an actor instance and our declIsolation is global actor
+        // isolated, but our context isolation is nonisolated... defer to flow
+        // isolation if this case passes the additional restrictions required by
```

Comment:
I think this bit in the comment needs some more work i.e. `deter to flow isolation for additional checking (e.x.: the initializer's nominal type has to be isolated).`